enclave/core/sgx/once.c

```diff
@@ -5,34 +5,39 @@
 #include <openenclave/internal/thread.h>
 #include <openenclave/internal/utils.h>
 
+#define SPINLOCK_NOT_USED 0
+#define SPINLOCK_ACTIVELY_HELD 1
+#define SPINLOCK_USED 2
+
 oe_result_t oe_once(oe_once_t* once, void (*func)(void))
 {
     if (!once)
         return OE_INVALID_PARAMETER;
 
     /* Double checked locking (DCLP). */
-    oe_once_t o = *once;
-
     /* DCLP Acquire barrier. */
     OE_ATOMIC_MEMORY_BARRIER_ACQUIRE();
-    if (o == 0)
+    if (*once != SPINLOCK_USED)
```

Comment:
"Use an atomic-acquire load operation to check if the function has not been invoked.
If the function has already been invoked, there is nothing to do.
If the function is being invoked, then this thread must wait for the function invocation to complete.
Otherwise, this thread can try to take ownership of invoking the function".